src/Security/Validation/AbstractPasswordValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SilverStripe\Security\Validation;
+
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberPassword;
+
+/**
+ * Abstract validator with functionality for checking for reusing old passwords.
+ */
+abstract class PasswordValidator
+{
+    use Injectable;
+    use Configurable;
+
+    /**
+     * Default number of previous passwords to check for a reusing old passwords.
+     */
+    private static int $historic_count = 6;
+
+    protected ?int $historicalPasswordCount = null;
+
+    public function validate(string $password, Member $member): ValidationResult
+    {
+        $result = ValidationResult::create();
+
+        $historicCount = $this->getHistoricCount();
+        if ($historicCount) {
+            $idColumn = DataObject::getSchema()->sqlColumnForField(MemberPassword::class, 'MemberID');
+            $previousPasswords = MemberPassword::get()
+                ->where([$idColumn => $member->ID])
+                ->sort(['Created' => 'DESC', 'ID' => 'DESC'])
+                ->limit($historicCount);
+            foreach ($previousPasswords as $previousPassword) {
+                if ($previousPassword->checkPassword($password)) {
+                    $error = _t(
+                        PasswordValidator::class . '.PREVPASSWORD',
+                        'You\'ve already used that password in the past, please choose a new password'
+                    );
+                    $result->addError($error, 'bad', 'PREVIOUS_PASSWORD');
```

Comment:
```suggestion
                    $result->addError($error, 'bad');
```
Don't need a code for uniqueness since we're we have `break` below

Reply:
That code was already there. I'd prefer to keep it give it and the break were both already there - maybe there's a good reason for it that we just don't have context for.